### ISSUE TITLE
Bug in shutdown controller when @filter_workers is nil

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#4271]
if there are no filter, the shutdown controller crashes:

```
% bin/logstash -e "" 
Settings: Default filter workers: 2
Logstash startup completed
^CSIGINT received. Shutting down the pipeline. {:level=>:warn}
The error reported is: 
  ["/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/pipeline.rb:374:in `plugin_threads'", "/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/pipeline.rb:365:in `stalling_threads'", "/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/shutdown_controller.rb:102:in `from_pipeline'", "/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/shutdown_controller.rb:50:in `start'", "org/jruby/RubyProc.java:281:in `call'", "/Users/joaoduarte/projects/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.22/lib/stud/interval.rb:20:in `interval'", "/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/shutdown_controller.rb:49:in `start'", "/Users/joaoduarte/projects/logstash/logstash-core/lib/logstash/shutdown_controller.rb:38:in `start'"] {:level=>:fatal}
%
```

**Comment by andrewvc** [comment]
Will this still be an issue in #4254 ?

**Comment by jsvd** [comment]
Not sure, it's a matter of checking the code path when not using filters.
From my understanding of the silver pipeline the filter+output thread count will always be > 0 so this problem shouldn't happen.

**Comment by jsvd** [comment]
closed by #4284
